platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/DataFileCompactor.java

```diff
@@ -0,0 +1,456 @@
+    /**
+     * The list of new files created during compaction. Usually, all files to process are compacted
+     * to a single new file, but if compaction is interrupted by a snapshot, there may be more than
+     * one file created.
+     */
+    private final List<Path> newCompactedFiles = new ArrayList<>();
+
+    /**
+     * Indicates whether compaction is in progress at the time when {@link #pauseCompaction()}
+     * is called. This flag is then checked in {@link DataFileCompactor#resumeCompaction()} )} to start a new
+     * compacted file or not.
+     */
+    private final AtomicBoolean compactionWasInProgress = new AtomicBoolean(false);
+
+    /**
+     * This variable keeps track of the compaction level that was in progress at the time when it was suspended.
+     * Once the compaction is resumed, this level is used to start a new compacted file, and then it's reset to 0.
+     */
+    private final AtomicInteger compactionLevelInProgress = new AtomicInteger(0);
+
+    /** When was the last medium-sized merge, only touched from single merge thread. */
+    private Instant lastMediumMerge;
+
+    /** When was the last full merge, only touched from single merge thread. */
+    private Instant lastFullMerge;
+
+    /** A nanosecond-precise Clock */
+    private final NanoClock clock = new NanoClock();
+
+    public DataFileCompactor(final DataFileCollection<?> dataFileCollection) {
+        this.dataFileCollection = dataFileCollection;
+        // Compute initial merge periods to a randomized value of now +/- 50% of merge period. So
+        // each node will do
+        // medium and full merges at random times.
+        lastMediumMerge = Instant.now()
+                .minus(config.mediumMergePeriod() / 2, config.mergePeriodUnit())
+                .plus((long) (config.mediumMergePeriod() * Math.random()), config.mergePeriodUnit());
+        lastFullMerge = Instant.now()
+                .minus(config.fullMergePeriod() / 2, config.mergePeriodUnit())
+                .plus((long) (config.fullMergePeriod() * Math.random()), config.mergePeriodUnit());
+    }
+
+    /**
+     * Compacts all files in filesToCompact.
+     *
+     * @param index          takes a map of moves from old location to new location. Once it is finished and
+     *                       returns it is assumed all readers will no longer be looking in old location, so old files
+     *                       can be safely deleted.
+     * @param filesToCompact   list of files to compact
+     * @return list of files created during the compaction
+     * @throws IOException          If there was a problem with the compaction
+     * @throws InterruptedException If the compaction thread was interrupted
+     */
+    // visible for testing
+    synchronized List<Path> compactFiles(final CASableLongIndex index, final List<DataFileReader<?>> filesToCompact)
+            throws IOException, InterruptedException {
+        if (filesToCompact.size() < 2) {
```

Comment:
As far as I remember, there is a config `minNumberOfFilesInMerge` to be used here instead of hard-coded constant of 2

Reply:
Fixed